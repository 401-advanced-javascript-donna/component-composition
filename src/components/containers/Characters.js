import React, { Component } from 'react';
import DeckOfCards from '../DeckOfCards';
import PropTypes from 'prop-types';
import apiCall from '../../services/api-call';

class Characters extends Component {
  static propTypes = {
    charCount: PropTypes.number.isRequired
  }
  state= {
    characters: [],
    pageCount: 1
  }

  componentDidMount() {
    apiCall('https://hey-arnold-api.herokuapp.com/api/v1/characters', this.props.charCount, this.state.pageCount)
      .then(result => this.setState({ characters: result }));
  }

  decrementPage = () => {
    this.setState(state => ({
      pageCount: state.pageCount - 1
    }), () => {
      apiCall('https://hey-arnold-api.herokuapp.com/api/v1/characters', this.props.charCount, this.state.pageCount)
        .then(result => this.setState({ characters: result }));
    });
  }

  incrementPage = () => {
    this.setState(state => ({
      pageCount: state.pageCount + 1
    }), () => {
      apiCall('https://hey-arnold-api.herokuapp.com/api/v1/characters', this.props.charCount, this.state.pageCount)
        .then(result => this.setState({ characters: result }));
    });
  }

  render() {
    return (
      <>
        <DeckOfCards items={this.state.characters} title='Characters' />
        <button disabled={this.state.pageCount === 1 ? true : false} onClick={this.decrementPage}>←</button>
        <button disabled={this.state.pageCount === 500 ? true : false} onClick={this.incrementPage}>→</button>
      </>
    );
  }
}

export default Characters;

