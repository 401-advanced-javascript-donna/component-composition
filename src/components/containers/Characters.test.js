import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters container', () => {
  global.fetch = jest.fn(() => Promise.resolve({}));
  it('renders Characters', () => {
    const wrapper = shallow(<Characters charCount={20} />);
    expect(wrapper).toMatchSnapshot();
  });
});
