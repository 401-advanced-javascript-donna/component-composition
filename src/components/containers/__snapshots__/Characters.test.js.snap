// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters container renders Characters 1`] = `
<Fragment>
  <DeckOfCards
    items={Array []}
    title="Characters"
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    ←
  </button>
  <button
    disabled={false}
    onClick={[Function]}
  >
    →
  </button>
</Fragment>
`;
